DSCResources/MSFT_xWebSiteDefaults/MSFT_xWebSiteDefaults.psm1

```diff
@@ -0,0 +1,176 @@
+	(
+        [Parameter(Mandatory)]
+        [ValidateSet("Machine")]
+        [string]$ApplyTo
+	)
+	
+    # Check if WebAdministration module is present for IIS cmdlets
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
+
+    $getTargetResourceResult = @{LogFormat = (GetValue "siteDefaults/logFile" "logFormat")
+                                    TraceLogDirectory = ( GetValue "siteDefaults/traceFailedRequestsLogging" "directory")
+                                    DefaultApplicationPool = (GetValue "applicationDefaults" "applicationPool")
+                                    AllowSubDirConfig = (GetValue "virtualDirectoryDefaults" "allowSubDirConfig")
+                                    ApplyTo = "Machine"
+                                    LogDirectory = (GetValue "siteDefaults/logFile" "directory")}    
+	return $getTargetResourceResult
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function will change a default setting if not already set
+######################################################################################
+function Set-TargetResource
+{
+	param
+	(	
+        [ValidateSet("Machine")]
+        [parameter(Mandatory = $true)]
+        [string]$ApplyTo,
+        [ValidateSet("W3C","IIS","NCSA","Custom")]
+        [string]$LogFormat,
+        [string]$LogDirectory,
+        [string]$TraceLogDirectory,
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+    )
+
+        CheckIISPoshModule
+
+        SetValue "siteDefaults/logFile" "logFormat" $LogFormat
+        SetValue "siteDefaults/logFile" "directory" $LogDirectory
+        SetValue "siteDefaults/traceFailedRequestsLogging" "directory" $TraceLogDirectory
+        SetValue "applicationDefaults" "applicationPool" $DefaultApplicationPool
+        SetValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test whether all given values are already set in the current configuration
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+        [ValidateSet("Machine")]
+        [parameter(Mandatory = $true)]
+        [string]$ApplyTo,
+        [ValidateSet("W3C","IIS","NCSA","Custom")]
+        [string]$LogFormat,
+        [string]$LogDirectory,
+        [string]$TraceLogDirectory,
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+	)
+
+    [bool]$DesiredConfigurationMatch = $true;
+
+    CheckIISPoshModule
+
+    $DesiredConfigurationMatch = CheckValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+    if (!($DesiredConfigurationMatch)) { return $false }
```

Comment:
I'd just change this line to:

    if (!(CheckValue -Path "virtualDirectoryDefaults" -Name "allowSubDirConfig" -NewValue $AllowSubDirConfig))
    {
        return $false
    }

No real need to assign the variable to just overwrite it two lines later.